src/Features/Core/Portable/CodeRefactorings/InvertIf/AbstractInvertIfCodeRefactoringProvider.Negator.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.InvertIf
+{
+    internal abstract partial class AbstractInvertIfCodeRefactoringProvider
+    {
+        internal static class Negator
+        {
+            private const string LongLength = "LongLength";
+
+            private static readonly Dictionary<BinaryOperatorKind, BinaryOperatorKind> s_negatedBinaryMap =
+                new Dictionary<BinaryOperatorKind, BinaryOperatorKind>
+                {
+                    {BinaryOperatorKind.Equals, BinaryOperatorKind.NotEquals},
+                    {BinaryOperatorKind.NotEquals, BinaryOperatorKind.Equals},
+                    {BinaryOperatorKind.LessThan, BinaryOperatorKind.GreaterThanOrEqual},
+                    {BinaryOperatorKind.GreaterThan, BinaryOperatorKind.LessThanOrEqual},
+                    {BinaryOperatorKind.LessThanOrEqual, BinaryOperatorKind.GreaterThan},
+                    {BinaryOperatorKind.GreaterThanOrEqual, BinaryOperatorKind.LessThan},
+                    {BinaryOperatorKind.Or, BinaryOperatorKind.And},
+                    {BinaryOperatorKind.And, BinaryOperatorKind.Or},
+                    {BinaryOperatorKind.ConditionalOr, BinaryOperatorKind.ConditionalAnd},
+                    {BinaryOperatorKind.ConditionalAnd, BinaryOperatorKind.ConditionalOr},
+                };
+
+            internal static SyntaxNode Negate(
+                SyntaxNode expression,
```

Comment:
feels like this can be generic over a TExpressionSyntax type, which may help other parts of the code.

Reply:
(wontfix)